extensions/adobe/experience/customerJourneyManagement/message-experienceevent.schema.json

```diff
@@ -0,0 +1,110 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/experience/customerJourneyManagement/message-experienceevent",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Adobe CJM ExperienceEvent Template Mixin",
+  "type": "object",
+  "description": "Adobe CJM ExperienceEvent mixin for ingesting the ExperienceEvent. Includes the core/standard ExperienceEvent as well as the required Adobe CJM ExperienceEvent mixins.",
+  "meta:extensible": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "meta:extends": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "definitions": {
+    "messageExperienceevent": {
+      "properties": {
+        "https://ns.adobe.com/xdm/experience/customerJourneyManagement/messageExecution": {
+          "title": "Message Execution",
+          "$ref": "https://ns.adobe.com/experience/customerJourneyManagement/messageexecution",
+          "description": "MessageExecution details for the Adobe CJM ExperienceEvent."
+        },
+        "https://ns.adobe.com/xdm/experience/customerJourneyManagement/messageProfile": {
+          "title": "Message Profile",
+          "$ref": "https://ns.adobe.com/experience/customerJourneyManagement/messageprofile",
+          "description": "MessageProfile details for the Adobe CJM ExperienceEvent."
+        },
+
+        "https://ns.adobe.com/xdm/experience/customerJourneyManagement/channelContext": {
+          "title": "Message Channel context",
+          "description": "Message Channel fields for the Messageprofle associated with the Adobe CJM Experience Event.",
+          "oneOf": [
```

Comment:
oneOf is not usable in AEP - you cannot have a field be two different data types.

Reply:
@cmathis : removed the one of construct from here, but still using under the "required" properties to apply a constraint that "channelContext" can have either "emailChannelContext" or "pushChannelContext"